Main Cleanup of Diffraction release notes

sf1919 opened the pull request:
Reordered, notes amalgamated where appropriate, spelling & grammar check, screenshot added.

**Description of work.**
The main refactoring the the Diffraction release notes ahead of release of 6.2. Includes:-

- Amalgamating notes where appropriate
- Adding/fixing links
- Re-ordering information
- Spelling & Grammar check
- Adding a screenshot


**To test:**
- remove ```build/docs/doctrees```
- build the documents again and check there are no warnings
- Check that the links work
- Check the information is still correct
- Are the images visible and clearly illustrate the new feature/improvement?
- Grammar/Spelling check

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Major refactoring of diffraction release notes

Reordered, notes amalgamated where appropriate, spelling & grammar check, screenshot added.

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
 - Fix the issue with the calibration diagnostics script when dealing with instruments of which the detector ID does not start from 0.
 - Fix the issue with :ref:`SNSPowderReduction <algm-SNSPowderReduction>` - when invalid height unit is encountered while reading sample log, we should continue by ignoring geometry and rely purely on user input.
-- fix d-spacing calculation when parabolic model is selected.
+- Fix d-spacing calculation when parabolic model is selected.
```
This is a bit confusingly worded - I believe it refers to a bug when converting d-spacing to TOF using diffractometer constants with non-zero DIFA - but it might be worth checking the original PR that added this (@DannyHindson might be able to confirm).

sf1919 replied:
I've found the PR - #31734. @KedoKudo can you confirm that @RichardWaiteSTFC is correct that this is about a bug when converting d-spacing to TOF using diffractometer constants with non-zero DIFA?

RichardWaiteSTFC replied:
Yep so having had a look at the PR (thanks for finding that!) it is indeed to fix conversion from TOF to dSpacing when DIFA non-zero

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
-- New setting for default peak function to fit in the Engineering Diffraction interface (initial default is :ref:`BackToBackExponential <func-BackToBackExponential>`).
-- Added serial fit capability to fitting tab in EngDiff UI - this fits all loaded workspaces with same initial parameters.
+- New setting for default peak function to fit in the :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>` (initial default is :ref:`BackToBackExponential <func-BackToBackExponential>`).
+- Added serial fit capability to :ref:`Fitting tab <ui engineering fitting>` in :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>` - this fits all loaded workspaces with same initial parameters.
```
nice addition of references to individual tabs in the UI docs (though I haven't built to check they work yet)!

sf1919 replied:
This will definitely need checking at some point as the Fitting Tab and Focus Tab links are brand new.

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
+- Sequential fitting in the :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>` now uses the output of the last successful fit (as opposed to the previous fit) as the initial parameters for the next fit.
+- An empty :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>` is no longer saved if the user saves a project having previously had the interface open at some point in that session.
+- The help button on the :ref:`Engineering Diffraction interface<Engineering_Diffraction-ref>` points to the correct page, having been broken in the last release.
+- Using the Clear button on the Workspace widget while using the :ref:`Fitting tab <ui engineering fitting>` no longer causes issues when you try to load runs back in.
```
Not sure about the use of `issues` here - if we can't be more specific perhaps `errors` - I will try and dif out the PR this relates to.

sf1919 replied:
I'm struggling to find the PR this is linked to. The release notes were added as part of #32284 - @jmborr do you know which PR originally fixed the error caused by the Clear button?

jmborr replied:
No, I'm afraid I don't. A lot of stuff was included in  #32284 that I had not coded for and unfortunately I don't know why 😞

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
+- New algorithm :ref:`HB3AIntegrateDetectorPeaks <algm-HB3AIntegrateDetectorPeaks>` for integrating four-circle data from HB3A in detector space.
+- New plotting script that provides diagnostic plots of :ref:`SCDCalibratePanels <algm-SCDCalibratePanels-v2>` on a per panel/bank basis.
+- Exposed :meth:`mantid.api.IPeak.getCol` and :meth:`mantid.api.IPeak.getRow` to python.
+- Added two integration methods to :ref:`HB3AIntegrateDetectorPeaks <algm-HB3AIntegrateDetectorPeaks>` for simple cuboid integration with and without fitted background.
```
Maybe put the two notes relating to the `HB3AIntegrateDetectorPeaks` together - or even better see if they can be combined. e.g. 

New algorithm :ref:`HB3AIntegrateDetectorPeaks <algm-HB3AIntegrateDetectorPeaks>` for integrating four-circle data from HB3A in detector space using simple cuboid integration with and without fitted background.

This is not for ISIS instruments though so perhaps it's best to check the above with @zjmorgan ?

zjmorgan replied:
That is correct. It is for HFIR only.

sf1919 replied:
I have combined the two notes in the first revision commit. Please can you check that it is correct @zjmorgan ?

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
+	- Documentation has been added as a guidance for general users.
+
+- :ref:`SCDCalibratePanels <algm-SCDCalibratePanels-v2>` major interface update along with enabling the calibration of T0 and sample position.
+- :ref:`SCDCalibratePanels <algm-SCDCalibratePanels-v2>` minor interface update that allows fine control of bank rotation calibration.
```
The `SCDCalibratePanels` should be moved to the single crystal section - and possibly this comment relating to bank rotation is already duplicated under single crystal improvements?

sf1919 replied:
I have changed the release notes so there is a whole section on ```SCDCalibratePanels``` in the Single Crystal improvements. 

@KedoKudo please can you check the revised notes are correct? Can you also check if the note ```New plotting script that provides diagnostic plots of :ref:`SCDCalibratePanels <algm-SCDCalibratePanels-v2>` on a per panel/bank basis.``` in the new features should be there? Or is it now covered in the Improvements section?

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
-- Existing :ref:`IntegrateEllipsoids <algm-IntegrateEllipsoids-v1>` now can use a different integrator for satellite peaks.
+
+- Existing :ref:`DGSPlanner <dgsplanner-ref>` expanded to support WAND².
+- Existing algorithm :ref:`IntegrateEllipsoids <algm-IntegrateEllipsoids-v1>` now can use a different integrator for satellite peaks.
```
Could you put the two notes relating to IntegrateEllipsoids adjacent to each other?

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,117 @@ Diffraction Changes
-- :ref:`LoadWANDSCD <algm-LoadWANDSCD-v1>` now has a new option to perform normalization in the same loading process.
+  on a scatter plot, new option for self-attenuation treatment using measured transmission.
+- Several improvements have been made to the group calibration routine including
+
```
Are these blank lines deliberate?

sf1919 replied:
Yes - its part of the formatting required for sublists. It also turns out you need to change - to * as we found out in the SANs release notes.... It will be correct in the next commit :)

sf1919 committed:
First set of revisions after review

Updated sublist formatting, reworded some notes on saving projects, SCDCalibratePanels and HB3AIntegrateDetectorPeaks. Moved notes where requested.

RichardWaiteSTFC commented:
Just had a look at the docs on the package you sent yesterday (i.e. before the latest changes). The links to the tabs in the fitting interface work really well thanks. I did notice however that the image added for the file filters (docs/source/images/EngDiff_Fit_Browse_Filters.png) looks a bit elongated (on windows non-4K screen)?
![image](https://user-images.githubusercontent.com/55979119/131974555-4d15c1a3-ab51-48e6-9375-8206097c2520.png)

sf1919 committed:
Second revision of notes after review

Further changes made based on comments plus some rewording for continuity e.g. changing Add to Added in line with other notes.

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,124 @@ Diffraction Changes
+* Existing :ref:`DGSPlanner <dgsplanner-ref>` expanded to support WAND².
+* Existing algorithm :ref:`IntegrateEllipsoids <algm-IntegrateEllipsoids-v1>` now can use a different integrator for satellite peaks.
+* New option in :ref:`IntegrateEllipsoids <algm-IntegrateEllipsoids-v1>` to share Bragg peak background with satellite peaks.
+* Existing algorithm:ref:`MaskPeaksWorkspace <algm-MaskPeaksWorkspace-v1>` now also supports tube-type detectors used at the CORELLI instrument.
```
think missing a space between algorithm and :ref:  - might explain why link isn't picked up
![image](https://user-images.githubusercontent.com/55979119/131996985-e3a78ff4-2b9a-4b6a-9aa6-f65fda1fb780.png)

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,124 @@ Diffraction Changes
+
+.. image::  ../../images/EngDiff_Fit_Browse_Filters.png
+   :align: center
+   :height: 700px
```
Can we make this slightly smaller - say 400px?

RichardWaiteSTFC commented on docs/source/release/v6.2.0/diffraction.rst:
```diff
@@ -5,110 +5,124 @@ Diffraction Changes
+  * Option to display all measured points on a scatter plot.
+  * New option for self-attenuation treatment using measured transmission.
+
+* Several improvements have been made to the group calibration routine including
```
Looks like there's a difference in formatting between the sub-bullet points lists on these two release notes 
![image](https://user-images.githubusercontent.com/55979119/131997619-cfc6f5b1-2256-47d6-9491-5d065f60f3c9.png)

RichardWaiteSTFC reviewed (changes requested):
Some small changes (hopefully quick) and then it can be merged. Good job on the first-pass, and the links to the EngDiff UI tabs work well.

sf1919 committed:
Minor revisions on formal review

sf1919 committed:
Attempt to format sublists correctly

RichardWaiteSTFC reviewed (approved):
Built fine, thanks for those few last changes. Looks good!

RichardWaiteSTFC commented:
@mantidproject/gatekeepers please can this be merged before there are any conflicts, thanks!
